coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala

```diff
@@ -21,6 +21,13 @@ import filodb.prometheus.ast.Vectors.PromMetricLabel
 import filodb.query.{exec, _}
 import filodb.query.exec._
 
+
+/**
+  * This is a container for any state to be passed up to parent nodes for easier query planning.
```

Comment:
Ew. This design needs a bit more work if it's expected to evolve into a general purpose collection of "hints" and "backdoor" state propagation.

Reply:
Introduced PlanResult.